File: iOS_runtime-archive/runtime-archive/runtime-archive/MomentModel.m
//
//  MomentModel.m
//  runtime-archive
//
//  Created by 孟跃平 on 2017/7/17.
//  Copyright © 2017年 com.mengyueping.www. All rights reserved.
//

#import "MomentModel.h"

@implementation MomentModel
- (void)encodeWithCoder:(NSCoder *)coder
{
    [coder encodeObject:_content forKey:@"_content"];
    [coder encodeObject:_time forKey:@"_time"];
    [coder encodeObject:_pic forKey:@"_pic"];
    [coder encodeObject:_place forKey:@"_place"];
}
- (instancetype)initWithCoder:(NSCoder *)coder
{
    self = [super init];
    if (self) {
        _content = [coder decodeObjectForKey:@"_content"];
        _time = [coder decodeObjectForKey:@"_time"];
        _pic = [coder decodeObjectForKey:@"_pic"];
        _place = [coder decodeObjectForKey:@"_place"];
    }
    return self;
}
@end

File: iOS_runtime-archive/swift-runtime-archive/swift-runtime-archive/MomentModel.swift
//
//  MomentModel.swift
//  swift-runtime-archive
//
//  Created by 孟跃平 on 2017/7/17.
//  Copyright © 2017年 com.mengyueping.www. All rights reserved.

// 一般归档

import UIKit

class MomentModel: NSObject, NSCoding {
    var content: String?
    var time: String?
    var pic: String?
    var place: String?
    
    override init() {
        super.init()
    }
    
    // MARK: - 归档
    func encode(with aCoder: NSCoder) {
        aCoder.encode(content, forKey: "content")
        aCoder.encode(time, forKey: "time")
        aCoder.encode(pic, forKey: "pic")
        aCoder.encode(place, forKey: "place")
    }
    // MARK: - 解档
    required init?(coder aDecoder: NSCoder) {
        super.init()
        
        content = aDecoder.decodeObject(forKey: "content") as? String
        time = aDecoder.decodeObject(forKey: "time") as? String
        pic = aDecoder.decodeObject(forKey: "pic") as? String
        place = aDecoder.decodeObject(forKey: "place") as? String
    }
}

